core/modules/filter/filter.module

```diff
@@ -725,8 +739,7 @@ function filter_formats($account = NULL) {
         );
         $filter_format = (object) $filter_format;
 
-        // We don't include disabled formats.
-        if (!$filter_format->status) {
+        if (!$include_disabled && !$filter_format->status) {
```

Comment:
Shouldn't this be an OR instead?

Reply:
Nope, that'd mean all formats get ignored when `$include_disabled` is FALSE...